modules/shortcodes/js/jetpack-email-subscribe-block.js

```diff
@@ -0,0 +1,51 @@
+/* global wp */
+/* jshint esversion: 5, es3:false */
+const el = wp.element.createElement,
+	registerBlockType = wp.blocks.registerBlockType,
+	ServerSideRender = wp.components.ServerSideRender,
+	TextControl = wp.components.TextControl,
+	InspectorControls = wp.editor.InspectorControls;
+
+const fields = [
+	{ id: 'title', label: 'Title' },
+	{ id: 'email_placeholder', label: 'Placeholder' },
+	{ id: 'submit_label', label: 'Submit button label' },
+	{ id: 'consent_text', label: 'Consent text' },
+	{ id: 'processing_label', label: '"Processing" status message' },
+	{ id: 'success_label', label: 'Success status message' },
+	{ id: 'error_label', label: 'Error status message' },
+];
+
+registerBlockType( 'jetpack/email-subscribe', {
+	title: 'Email Subscribe Block',
+	icon: 'email',
+	category: 'widgets',
+
+	edit: function( props ) {
+		const edit_controls = fields.map( function( field ) {
+			return el( InspectorControls, {},
```

Comment:
`InspectoControls` should be moved outside of the loop. It can take multiple children elements. With that property to every `TextControl` should have `key` prop  to ensure that React reuses existing component when re-rendering, e.g.:
```js
key: field.id,
```

Reply:
Done, I moved InspectorControls to a hegher scope and refactored a bit.
I added `key` as well.